### PR TITLE
Settings `type` invalid defaultTo fix

### LINKED ISSUE #12181: Improve settings defaults
refs https://github.com/TryGhost/Ghost/issues/10318

## Problems
After [restructuring](https://github.com/TryGhost/Ghost/issues/10318) `settings` models and db structure there have been some leftover pieced which would need sorting. Specifically:
1. Current [settings model default](https://github.com/TryGhost/Ghost/blob/91bca9f109361e80991ce1cef2c910dcd069cf0d/core/server/models/settings.js#L95) for`type` field does not correspond to any of the values [expected by the schema](https://github.com/TryGhost/Ghost/blob/91bca9f109361e80991ce1cef2c910dcd069cf0d/core/server/data/schema/schema.js#L178-L184). 
2. Schema definition for [settings defaultsTo](https://github.com/TryGhost/Ghost/blob/91bca9f109361e80991ce1cef2c910dcd069cf0d/core/server/data/schema/schema.js#L176) is an invalid `core` value. Same as for pt. 1 `core` is not a valid value according to schema validations. 

## Solution
The default in both model and schema should be changed to one of the valid values. There are 2 possible solutions:
1. `string` - as that's the most used type for this column.
2. No default at all, so the insertion fails by default when there is no value. This is less used pattern - we only act similarly for `api_keys.type`.

The 1st option seemed to have most consensus when discussed within the team.

Additionally the model defaults for `settings` should be investigated a little deeper. The don't seem to ever kick in, so it is possible this piece of code can be removed entirely.

### PR DESCRIPTION
**Pull request by lunaticmonk**
- [x] There's a clear use-case for this code change
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test` and `yarn lint`)

Fixes https://github.com/TryGhost/Ghost/issues/12181

More info can be found by clicking the "guidelines for contributing" link above.

### CODE REVIEW

**Comment by lunaticmonk** [comment]
@naz Please review.

File: core/server/models/settings.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ Settings = ghostBookshelf.Model.extend({
 
     defaults: function defaults() {
         return {
-            type: 'core'
+            type: ''
```
Comment (naz) [inline core/server/models/settings.js:95]: Does this default ever gets executed? If it's not used I think we could remove the `defaults` function completely?

**Review by naz — CHANGES_REQUESTED** [review]

File: core/server/data/schema/schema.js [hunk 2]
```diff
@@ -173,7 +173,7 @@ module.exports = {
             type: 'string',
             maxlength: 50,
             nullable: false,
-            defaultTo: 'core',
+            defaultTo: '',
```
Comment (naz) [inline core/server/data/schema/schema.js:176]: Don't think there should be any default in this case at all. If it were to default to something it should at least be a value form `validations`. Removing `defaultTo: ''` parameter should be fine :+1:

**Review by naz — CHANGES_REQUESTED** [review]
The `defaultTo` value in schema should not be set to empty string :wink:

**Comment by lunaticmonk** [comment]
@naz Pardon for being late. Full time job keeping busy 😄 Will do it in a day!

**Comment by naz** [comment]
@lunaticmonk all good and no hurry :) Thanks for handling this!

**Comment by naz** [comment]
@lunaticmonk all good and no hurry :) Thanks for handling this!
@daniellockyer would be good to have your :heavy_check_mark:  on this ;)

**Review by daniellockyer — APPROVED** [review]

**Review by hc0503 — APPROVED** [review]

**Comment by naz** [comment]
This PR has landed in master as https://github.com/TryGhost/Ghost/commit/e756af65f2666926f1621cc50e4014dd74493959. Thanks for your contribution @lunaticmonk!

**Pull request closed without merging**